### PR TITLE
[Update] Added ERD to README

### LINKED ISSUE #21: [UPDATE] Update ERD Diagram
## Task Description
- Update to include a new table "Cancellation Types" to existing ERD as discussed in our recent project meeting.
- Modify the data types of specified columns in the ERD to align with the new database schema requirements.
- Other fixes mentioned during our meeting.
## User Stories
- As an engineer, I want to see an updated ERD, so I can implement the database changes accurately.
## Relevant Files [If Available]

### PR DESCRIPTION
**Pull request by isaacschaessenscoleman**
Here's the latest iteration of the ERD: 

[DrawSQLLink](https://drawsql.app/teams/isaac-11/diagrams/railway-tracker-3)

Screenshot 2024-04-24 at 10 46 32

## Changes include:
- Removal of ServiceType table and attribute because after a look through the RealTime Trains API, I've only found 'train', 'bus' and 'ship' services ... i think we're only concerned with trains for this project. 
- Added user table - main purpose: to allow for the subscription functionality
- Added AlertSubscriptionRecord - this table keeps a record of all of the operators that users are subscribed to for alerts
- Added SummarySubscriptionRecord - this table keeps a record of all of the stations that users are subscribed to for weekly reports

# What I need
- Does this ERD make sense to you? Do you have any questions?
- Do the datatypes and relationships make sense?
- StationSubRecord and OperatorSubRecord represent the tables for the subscriptions to the weekly report and alerts, respectively. Weekly reports are related to stations and alerts are related to operators. Do these table names make sense?
- Could there be better table names?
- Could there be better column names?

Closes #21

### CODE REVIEW

**Comment by adamski201** [comment]
Looks great - could we add scheduledArrival time to the cancellation table so that we a datetime associated with it? That way we know how many cancellations happened per day.

**Review by adamski201 — CHANGES_REQUESTED** [review]
Looks great - could we add scheduledArrival time to the cancellation table so that we a datetime associated with it? That way we know how many cancellations happened per day.

Subjective and just my opinion - we could rename OperatorSubRecord to be more clear, what do you think about just calling it OperatorSubscription? Similarly, StationSubscription.

File: README.md [hunk 1]
```diff
@@ -73,7 +73,12 @@ terraform destroy
 
 ### ERD diagram
 
-![ERD diagram](...)
+![ERD diagram](https://drawsql.app/teams/isaac-11/diagrams/railway-tracker-3)
+
+![image](https://github.com/adamski201/Railway-Tracker/assets/84942651/9e6dc207-77bf-41b4-b83a-ae77c5825deb)
```
Comment (SaniyaShaikhh) [inline README.md:78]: Looks really good!! very clear and consistent styling

**Comment by aribasyeda** [comment]
Great work Isaac, the ERD looks amazing. Just a few suggestions:
- we can possibility rename cancellation_record and arrivals_record to arrivals and cancellations for simplicity
- if renamed, consistent plurality 
- the column under 'Affected Routes' incident tables could be split further just to stick to being 3NF? just a suggestion 

Screenshot 2024-04-25 at 10 38 23

**Comment by aribasyeda** [comment]
Updated README

New Link: https://drawsql.app/teams/trainee-4/diagrams/copy-of-railway-tracker

Screenshot 2024-04-25 at 10 38 23

**Comment by adamski201** [comment]
Can we get a working link so I can just do a final check before merging? Thanks @aribasyeda

**Comment by aribasyeda** [comment]
Updated link: https://drawsql.app/teams/trainee-4/diagrams/railway-tracker-3nf

File: README.md (hunk 1 above)
Comment (isaacschaessenscoleman) [reply README.md]: Good idea - I've made those changes and pushed, so should be fixed now!

**Comment by isaacschaessenscoleman** [comment]
> Great work Isaac, the ERD looks amazing. Just a few suggestions:
> 
> * we can possibility rename cancellation_record and arrivals_record to arrivals and cancellations for simplicity
> * if renamed, consistent plurality
> * the column under 'Affected Routes' incident tables could be split further just to stick to being 3NF? just a suggestion
> * had a go at shuffling the tables (attached below)

- Good idea - Table names have been simplified and changes pushed.
- Consistent plurality of names is consistent in latest push.
- Could definitely be useful but there's a danger of added unnecessary complexity - we'll have a group meting about that later to whether it might be useful or not. 
- We will definitely be using that - I can be hopeless when it comes to clear visualisations. Thank you!!!

**Review by aribasyeda — CHANGES_REQUESTED** [review]
ERD link updated to https://drawsql.app/teams/trainee-4/diagrams/railway-tracker-3nf

**Review by aribasyeda — APPROVED** [review]
Amazing work ⭐ Thanks for updating!!

**Comment by adamski201** [comment]
Turns out PSQL doesn't take into account capitalisation so we'll need to change to using underscore to maintain readability.

https://giswiki.hsr.ch/SQL_and_PL/pgSQL_Programming_Style_Guide

**Comment by aribasyeda** [comment]
Good catch Adam! Lets create 2 tickets for updating the ERD and one for updating the schema script

**Pull request merged**